### PR TITLE
Calendar: show an action sheet when user long-presses calendar event

### PR DESCRIPTION
**Pull request by simonrozsival**
## What's this?

screen shot 2019-01-10 at 12 34 02 pm

### Relationships
Ref #3622 

## Why do we want this?
We want to have a more useful calendar view.

## How is it done?
The editing helper ignored a long press for non-editable (== calendar events) items. I added a new subject/observable which emits when there is a long press on some calendar event.

I had to modify the `IDialogService.Select` constraint to allow non-reference types. We only use it in the reports view and the callsite there is OK if we drop the constraint, so I guess it can't be a problem.

All the strings are translatable.

### Side effects
I hope I didn't break anything along the way (editing time entries). Well I am sure I didn't break anything and I couldn't find a bug in the simulator. The problem is that _sometimes_ I'm sure I didn't break anything but in fact I did... so please try playing with the calendar in the simulator/on your phone and try to break it.

## Review hints
☝️ 

## :squid: Permissions
🦑

### CODE REVIEW

**Comment by amulware** [comment]
Please add a screenshot above. :)

File: Toggl.Foundation/Resources.resx [hunk 1]
```diff
@@ -495,6 +495,18 @@ account is not set up</value>
     <data name="Calendar">
         <value>Calendar</value>
     </data>
+    <data name="CalendarWhatToDoWithCalendarEvent">
+        <value>What would you like to do with the calendar event?</value>
```
Comment (amulware) [inline Toggl.Foundation/Resources.resx]: `What would you like to do with this calendar event?`

File: Toggl.Foundation/Resources.resx [hunk 2]
```diff
@@ -495,6 +495,18 @@ account is not set up</value>
...
+    </data>
+    <data name="CalendarStartNow">
+        <value>Start now</value>
+    </data>
+    <data name="CalendarStartWhenTheEventStarts">
+        <value>Start at event start</value>
+    </data>
+    <data name="CalendarCopyEventToTimeEntry">
+        <value>Copy to time entry</value>
```
Comment (amulware) [inline Toggl.Foundation/Resources.resx]: `Copy as time entry`

**Comment by ApocalypticFish** [comment]
The copy looks good.

**Pull request merged**